src/slurm_plugin/clustermgtd.py

```
@@ -1126,36 +885,123 @@ def _handle_protected_mode_process(self, slurm_nodes, partitions, bootstrap_fail
+
+        instance_id_to_active_node_map = {}
+        inactive_instance_ids = []
+        orphaned_instances = []
+        for instance in cluster_instances:
+            if instance.private_ip in active_ip_to_slurm_node_map:
+                active_ip_to_slurm_node_map[instance.private_ip].instance = instance
+                instance_id_to_active_node_map[instance.id] = active_ip_to_slurm_node_map[instance.private_ip]
+            elif instance.private_ip in inactive_ip_to_slurm_node_map:
+                inactive_ip_to_slurm_node_map[instance.private_ip].instance = instance
+                inactive_instance_ids.append(instance.id)
+            else:
+                orphaned_instances.append(instance)
+
+        return instance_id_to_active_node_map, inactive_instance_ids, orphaned_instances
+
+    def _initialize_nodes(self):
+        """Initialzie slurm nodes and partitions."""
+        try:
+            log.info("Retrieving nodes info from the scheduler")
+            active_nodes, inactive_nodes = self._get_node_info_from_partition()
+            log.debug("Current active slurm nodes in scheduler: %s", active_nodes)
+            log.debug("Current inactive slurm nodes in scheduler: %s", inactive_nodes)
+            return active_nodes, inactive_nodes
+        except ClusterManager.SchedulerUnavailable:
+            log.error("Unable to get partition/node info from slurm, no other action can be performed. Sleeping...")
+            return
+
+    def _get_cluster_instances(self):
+        try:
+            # After reading Slurm nodes wait for 5 seconds to let instances appear in EC2 describe_instances call
+            time.sleep(5)
+            log.info("Retrieving list of EC2 instances associated with the cluster")
+            return self._get_ec2_instances()
+        except ClusterManager.EC2InstancesInfoUnavailable:
+            log.error("Unable to get instances info from EC2, no other action can be performed. Sleeping...")
+            return
```

We would want to either raise the exception for the caller to handle the exception or explicitly returning None